tsdb/db.go

```diff
@@ -621,6 +621,30 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 
 	var wlog *wal.WAL
 	segmentSize := wal.DefaultSegmentSize
+
+	defer func() {
+		// Close files if startup fails somewhere
+		if err != nil {
+			if wlog != nil {
+				if er := wlog.Close(); er != nil {
```

Comment:
```suggestion
				if wlog != nil && er := wlog.Close(); er != nil {
```

Reply:
This does not compile, it would compile if I did this:
```
if er := wlog.Close(); wlog != nil && er != nil {
```
But then I could try to close a nil wal

Reply:
oh sorry :) then 2 lines are fine I guess